### PR TITLE
add configuration description about isolation-read.engines & proxy-protocol.networks

### PR DESCRIPTION
**Pull request by pepezzzz**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

add configuration description about isolation-read.engines & proxy-protocol.networks

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
- https://docs.pingcap.com/zh/tidb/v5.0/command-line-flags-for-tidb-configuration#--proxy-protocol-networks
- https://github.com/pingcap/tidb/blob/master/config/config.toml.example#L488

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@pepezzzz Could you please involve one or two tech reviewers? Thanks~

**Comment by pepezzzz** [comment]
@leoppro  @bb7133  PTAL

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -705,6 +705,31 @@ TiDB 服务状态相关配置。
...
 
+## isolation-read
+
+读取隔离相关的配置项。
+
+### `engines`
+
++ 用于控制 TiDB 节点允许从哪种类型的引擎读取数据。
++ 默认值：["tikv", "tiflash", "tidb"]，表示由优化器自动选择存储引擎。
++ 可选值："tikv", "tiflash", "tidb" 的组合，如：["tikv", "tidb"]、["tiflash", "tidb"]。
```
Comment (TomShawn) [inline tidb-configuration-file.md:716]: @cfzjywxk PTAL. Thanks!

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -705,6 +705,31 @@ TiDB 服务状态相关配置。
...
+### `networks`
+
++ 允许使用 [PROXY 协议](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt)连接 TiDB 的代理服务器地址列表。
++ 默认值：""
++ 通常情况下，通过反向代理使用 TiDB 时，TiDB 会将反向代理服务器的 IP 地址视为客户端 IP 地址。对于支持 [PROXY 协议](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt)的反向代理（如 HAProxy），开启 PROXY 协议后能让反向代理透传客户端真实的 IP 地址给 TiDB。
++ 配置该参数后，TiDB 将允许配置的源 IP 地址使用 PROXY 协议连接到 TiDB，且拒绝这些源 IP 地址使用非 PROXY 协议连接。若该参数为空，则任何源 IP 地址都不能使用 PROXY 协议连接到 TiDB。地址可以使用 IP 地址格式 (192.168.1.50) 或者 CIDR 格式 (192.168.1.0/24)，并可用 `,` 分隔多个地址，或用 `*` 代表所有 IP 地址。
+
+> **警告：**
+>
+> 需谨慎使用 `*` 符号，因为它可能引入安全风险，允许来自任何 IP 的客户端自行汇报其 IP 地址。另外，它可能会导致部分直接连接 TiDB 的内部组件无法使用，例如 TiDB Dashboard。
```
Comment (TomShawn) [inline tidb-configuration-file.md:731]: @superlzs0476 @yiwen92 PTAL. Thanks!

File: tidb-configuration-file.md (hunk 1 above)
Comment (cfzjywxk) [reply tidb-configuration-file.md:716]: Looks ok.

**Review by superlzs0476 — APPROVED** [review]
LGTM

File: best-practices/haproxy-best-practices.md [hunk 3]
```diff
@@ -203,6 +203,18 @@ listen tidb-cluster                        # 配置 database 负载均衡。
    server tidb-3 10.9.64.166:4000 check inter 2000 rise 2 fall 3
 ```
 
+如要通过 show processlist 查看连接来源 IP，需要配置使用 [PROXY 协议](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt)连接 TiDB。
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:206]:
```suggestion
如要通过 `SHOW PROCESSLIST` 查看连接来源 IP，需要配置使用 [PROXY 协议](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt)连接 TiDB。
```

File: best-practices/haproxy-best-practices.md [hunk 4]
```diff
@@ -203,6 +203,18 @@ listen tidb-cluster                        # 配置 database 负载均衡。
...
+
+```yaml
+   server tidb-1 10.9.18.229:4000 send-proxy check inter 2000 rise 2 fall 3       
+   server tidb-2 10.9.39.208:4000 send-proxy check inter 2000 rise 2 fall 3
+   server tidb-3 10.9.64.166:4000 send-proxy check inter 2000 rise 2 fall 3
+```
+
+> **注意：**
+>
+> 使用 PROXY 协议，需要对应地添加 tidb server 的 `proxy-protocol.networks` 配置文件参数。
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:216]:
```suggestion
> 使用 PROXY 协议时，你需要在 tidb-server 的 `proxy-protocol.networks` 配置文件中添加对应的参数。
```

File: tidb-configuration-file.md [hunk 5]
```diff
@@ -705,6 +705,31 @@ TiDB 服务状态相关配置。
...
+
+## proxy-protocol
+
+PROXY 协议相关的配置项。
+
+### `networks`
+
++ 允许使用 [PROXY 协议](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt)连接 TiDB 的代理服务器地址列表。
++ 默认值：""
++ 通常情况下，通过反向代理使用 TiDB 时，TiDB 会将反向代理服务器的 IP 地址视为客户端 IP 地址。对于支持 [PROXY 协议](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt)的反向代理（如 HAProxy），开启 PROXY 协议后能让反向代理透传客户端真实的 IP 地址给 TiDB。
```
Comment (TomShawn) [inline tidb-configuration-file.md:726]:
```suggestion
+ 通常情况下，通过反向代理使用 TiDB 时，TiDB 会将反向代理服务器的 IP 地址视为客户端 IP 地址。对于支持 PROXY 协议的反向代理（如 HAProxy），开启 PROXY 协议后能让反向代理透传客户端真实的 IP 地址给 TiDB。
```

File: tidb-configuration-file.md (hunk 2 above)
Comment (TomShawn) [inline tidb-configuration-file.md:731]:
```suggestion
> 需谨慎使用 `*` 符号，因为 `*` 允许来自任何 IP 的客户端自行汇报其 IP 地址，从而可能引入安全风险。另外，`*` 可能导致部分直接连接 TiDB 的内部组件无法使用，例如 TiDB Dashboard。
```

**Review by cfzjywxk — APPROVED** [review]

**Review by TomShawn — APPROVED** [review]

**Pull request merged**